src/server/oasisapi/analyses/tasks.py

```diff
@@ -18,6 +19,57 @@
 logger = get_task_logger(__name__)
 
 
+class LogTaskError(Task):
+    # from gist https://gist.github.com/darklow/c70a8d1147f05be877c3
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        self.handle_task_failure(exc, task_id, args, kwargs, einfo)
+        super(LogTaskError, self).on_failure(exc, task_id, args, kwargs, einfo)
+
+    def handle_task_failure(self, exc, task_id, args, kwargs, traceback):
+        logger.info('name: {}'.format(self.name))
+        logger.info('args: {}'.format(args))
+        logger.info('kwargs: {}'.format(kwargs))
+        logger.info('traceback: {}'.format(traceback))
+
+        try:
```

Comment:
Wouldn't it be better to move the try-catch to the on_failure method? Makes this a little easier to read.